### PR TITLE
Places list. Fixes #154

### LINKED ISSUE #154: List of populated places
In original Blaxxun communities, there was a list of places that had people in them, and I believe a count of how many.

This would be convenient for people to see which places they can explore that have people in them.

### PR DESCRIPTION
**Pull request by Sgeo**

### CODE REVIEW

File: spa/src/components/Chat.vue [hunk 1]
```diff
@@ -184,9 +212,29 @@ export default Vue.extend({
...
     },
+    goToSlug(slug): void {
+      this.$router.push({ path: `/place/${slug}` });
+    },
     moveObject(objectId): void {
       this.$emit("move-object", objectId);
     },
+    async refreshPlaces(): Promise<void> {
+      const population: {[index: string]: number} = await fetch('/population').then(r => r.json());
+      console.log(population);
```
Comment (dburleson) [inline spa/src/components/Chat.vue:223]: suggest removing console log

File: spa/src/components/Chat.vue [hunk 2]
```diff
@@ -184,9 +212,29 @@ export default Vue.extend({
...
+      const placesInfo = await fetch(`/api/place/${Object.keys(population).join(',')}/ids`).then(r => r.json()).then(j => j.places);
+      const placeIdMapping = function(id) {
+        const idInt = parseInt(id, 10);
+        return placesInfo.filter(place => place.id === idInt)[0] ?? {name: "<Unknown>", slug: ""};
+      };
+      console.log(placesInfo);
```
Comment (dburleson) [inline spa/src/components/Chat.vue:229]: remove console log

File: spa/src/components/Chat.vue [hunk 3]
```diff
@@ -184,9 +212,29 @@ export default Vue.extend({
...
+      this.places = Object.entries(population)
+        .sort(([id_a, count_a], [id_b, count_b]) => count_b - count_a)
+        .map(([id, count]) => {
+          const placeInfo = placeIdMapping(id);
+          return {name: placeInfo.name, count: count, slug: placeInfo.slug}
+        });
+      console.log(this.places);
```
Comment (dburleson) [inline spa/src/components/Chat.vue:236]: remove console log

File: spa/server.js [hunk 4]
```diff
@@ -42,6 +42,20 @@ app.get('/', (req, res) => {
   res.sendFile(path.join(__dirname, "/dist/index.html"));
 });
 
+app.get('/population', (req, res) => {
```
Comment (dburleson) [inline spa/server.js:45]: @Sgeo Does this endpoint still work? I don't see it in the routes.

File: api/src/routes/place.routes.ts [hunk 5]
```diff
@@ -7,5 +7,7 @@ placeRoutes.get('/:slug/object_instance',
   (request, response) => placeController.getPlaceObjects(request, response));
 placeRoutes.get('/:slug',
   (request, response) => placeController.getPlace(request, response));
+placeRoutes.get('/:ids/ids',
```
Comment (dburleson) [inline api/src/routes/place.routes.ts:10]: @TheMikeSanto Feedback on this endpoint URL. Something about comma seperated ids in a URL makes me feel funny. But maybe that's just me.

**Review by dburleson — COMMENTED** [review]
Overall, not tested yet, but I like where this is going. Could be a really good addition to highlighting where active users are.

File: spa/server.js (hunk 4 above)
Comment (dburleson) [reply spa/server.js:45]: of course, ignore me. I was still in "api" mode when looking at this file.

**Comment by dburleson** [comment]
Finally got around to reviewing this a bit more and while this code does provide the feature intended, I think there is a better way to approach this feature which would also lay the foundation for future features.

Using a `session` table would be a better fit for this and would allow expansion into other future features such as "friends online", "users online" feature.

A session table would consist of fields such as member_id, place_id. Creating this table via a knex migration would include in the appropriate timestamps too.

An endpoint could be used such as `/members/online` which could then return an array along the lines of:

```
[
{member{}, place{}},
{member{}, place{}},
{member{}, place{}}
]
```

Everytime a user enterd a place, a record can be inserted/updated with their current place_id. It could then be pinged (every 3 minutes?) to keep up to date that the user is still there.

**Comment by TheMikeSanto** [comment]
@Sgeo here is some docs on how to create a database migration if you're going to be adding a new table: https://knexjs.org/guide/migrations.html#migration-cli

**Pull request closed without merging**